### ISSUE TITLE
pfQuest note integration

### ISSUE THREAD

**Issue by ReynoldsCahoon** [cralor/TourGuideVanilla#52]
![image](https://github.com/cralor/TourGuideVanilla/assets/2556251/60a9645e-124d-430f-bffd-fed97ca984a5)
_TomTom is the top arrow, pfQuest the bottom arrow._

I'm creating a custom guide, and I've been testing the optional navigation addons pfQuest and TomTomVanilla.

TomTom will properly route to my guides' notes, however pfQuest seems to only have quest tracking. Has anyone found a way to add pfQuest trackable objectives that are passed from notes in TourGuide?

Not only do I prefer the guidance arrow and text formatting in pfQuest, it also has smooth transitions and functionality that TomTom does not.

| Heading to quest objective | Arriving at quest objective |
| --- | --- |
| ![image](https://github.com/cralor/TourGuideVanilla/assets/2556251/b73a89cb-de43-4039-a823-40d4ff5d7d1b) | ![image](https://github.com/cralor/TourGuideVanilla/assets/2556251/d4fc69f5-f0c4-4c96-aec4-c41d8ea331e7) |

It also points to objectives in a useful way with context-appropriate icons over the navigation arrow.
![image](https://github.com/cralor/TourGuideVanilla/assets/2556251/caa7d9f8-df5b-47df-9570-4f42552606df)

TLDR: I think this addon would be enhanced significantly if the pfQuest navigation arrow could display TourGuide notes that aren't quest related (as TomTom does).

**Comment by cralor** [comment]
Neat. This would require tying in pfQuest's functionality to TourGuide's Mapping module.

Applicable files:
https://github.com/shagu/pfQuest/blob/60998f2bb1d063ef2341bf67d5147575988eda33/route.lua
https://github.com/shagu/pfQuest/blob/60998f2bb1d063ef2341bf67d5147575988eda33/map.lua#L816
https://github.com/cralor/TourGuideVanilla/blob/master/Mapping.lua

Upon quick glance, it doesn't appear pfQuests arrow has an open-ended function for quick use. But I'm sure something could be figured out.

**Comment by shagu** [comment]
I've added a small example on how to create waypoints and how to pin them in the linked issue above.

However you should keep in mind that TomTom was made for other addons, more like a library.
Mine on the other hand is just a hacky solution built to fulfil my own needs and tightly integrated into my code.

I'm open and willing to support API calls for it if something is needed, however I will not rewrite big parts for it.
So let's see how far we can get. Let me know if there are questions or things that **must** be changed in pfQuest in order to get your stuff done.

PS: Also nice to see you're still around @cralor :)

**Comment by cralor** [comment]
Wow, you went above and beyond here @shagu :) Much appreciated. I haven't been in this space very much, but I have really enjoyed developing this and incorporated pfQuest in it as well. And got some heavy lifting with Road-block and others of course!
